documents/read_acls.go

```diff
@@ -113,13 +113,13 @@ func (cd *CoreDocument) addNFTToReadRules(registry common.Address, tokenID []byt
 	role := newRole()
 	role.Nfts = append(role.Nfts, nft)
 	cd.addNewRule(role, coredocumentpb.Action_ACTION_READ)
-	return cd.setSalts()
+	return nil
 }
 
 // AddNFT returns a new CoreDocument model with nft added to the Core Document. If grantReadAccess is true, the nft is added
 // to the read rules.
 func (cd *CoreDocument) AddNFT(grantReadAccess bool, registry common.Address, tokenID []byte) (*CoreDocument, error) {
-	ncd, err := cd.PrepareNewVersion(nil, false)
+	ncd, err := cd.PrepareNewVersion(nil)
```

Comment:
lets change the function definition of PrepareNewVersion to take variardic input instead of slice. Then the call would be 

```
cd.PrepareNewVersion() // with no collaborators

cd.PrepareNewVersion(collaborator) // with one collaborator

cd.PrepareNewVerison(collaborators...) // with multiple collaborators
```